include/swift/AST/Decl.h

```diff
@@ -2207,6 +2207,15 @@ class ValueDecl : public Decl {
     return result;
   }
 
+  /// Determine whether this Decl has either Private or FilePrivate access.
+  bool isPrivateOrFilePrivate() const {
```

Comment:
This is *still* not great because `getFormalAccess` is something like "get access as written", which means this is really asking if something is *explicitly* declared private or fileprivate. For finding the outermost non-local private scope that's fine, but it's not a query we generally want to expose; the *only* place it's interesting is for mangling and then the debug info to support that mangling.

So maybe that's the right level of abstraction: `isOutermostPrivateOrFileprivateScope`.